drivers/flash/spi_nor.c

```diff
@@ -166,10 +227,14 @@ static int spi_nor_read(struct device *dev, off_t addr, void *dest,
 
 	SYNC_LOCK();
 
+	(void)exit_dpd(dev);
```

Comment:
Better implement  device power management for the driver instead of putting the device in low power mode for each transaction and leave the decision to put the device in low power mode to the application code.

Reply:
I'd love to, but it's useless.  There's no way to propagate the dependency information down to the driver unless you're interacting with the flash directly.

Reply:
@vanwinkeljan Please re-check this.

I've added a Kconfig option to allow control of whether deep power-down is entered between transactions.  This should preserve existing behavior, while still allowing my use case of dropping current consumption in applications which rarely use the flash.

If you absolutely insist I will go ahead and implement the device power management (eventually), but as I note in the commit message and above it's not every useful until we get device dependency in place so that clients of the device can ensure it's adequately woken before it's used.  As it stands many applications won't even know (or care) whether their file system is on a SPI NOR device, or how to access that device to control its power consumption.

Reply:
Thx for the update, adding the Kconfig is fine for me.